packages/in_app_purchase/lib/src/billing_client_wrappers/sku_details_wrapper.dart

```diff
@@ -0,0 +1,114 @@
+  /// types of all of the members on this class.
+  static SkuDetailsWrapper fromMap(Map<String, dynamic> map) =>
+      SkuDetailsWrapper(
+          description: map['description'],
+          freeTrialPeriod: map['freeTrialPeriod'],
+          introductoryPrice: map['introductoryPrice'],
+          introductoryPriceMicros: map['introductoryPriceMicros'],
+          introductoryPriceCycles: map['introductoryPriceCycles'],
+          introductoryPricePeriod: map['introductoryPricePeriod'],
+          price: map['price'],
+          priceAmountMicros: map['priceAmountMicros'],
+          priceCurrencyCode: map['priceCurrencyCode'],
+          sku: map['sku'],
+          subscriptionPeriod: map['subscriptionPeriod'],
+          title: map['title'],
+          type: SkuType.fromString(map['type']),
+          isRewarded: map['isRewarded']);
+
+  final String description;
+  final String freeTrialPeriod;
+  final String introductoryPrice;
+  final String introductoryPriceMicros;
+  final String introductoryPriceCycles;
+  final String introductoryPricePeriod;
+  final String price;
+  final int priceAmountMicros;
+  final String priceCurrencyCode;
+  final String sku;
+  final String subscriptionPeriod;
+  final String title;
+  final SkuType type;
+  final bool isRewarded;
+
+  @override
+  bool operator ==(dynamic other) =>
+      other is SkuDetailsWrapper &&
+      other.description == description &&
+      other.freeTrialPeriod == freeTrialPeriod &&
+      other.introductoryPrice == introductoryPrice &&
+      other.introductoryPriceMicros == introductoryPriceMicros &&
+      other.introductoryPriceCycles == introductoryPriceCycles &&
+      other.introductoryPricePeriod == introductoryPricePeriod &&
+      other.price == price &&
+      other.priceAmountMicros == priceAmountMicros &&
+      other.sku == sku &&
+      other.subscriptionPeriod == subscriptionPeriod &&
+      other.title == title &&
+      other.type == type &&
+      other.isRewarded == isRewarded;
+
+  @override
+  int get hashCode =>
+      description.hashCode +
+      freeTrialPeriod.hashCode +
+      introductoryPrice.hashCode +
+      introductoryPriceMicros.hashCode +
+      introductoryPriceCycles.hashCode +
+      introductoryPricePeriod.hashCode +
+      price.hashCode +
+      priceAmountMicros.hashCode +
+      sku.hashCode +
+      subscriptionPeriod.hashCode +
+      title.hashCode +
+      type.hashCode +
+      isRewarded.hashCode;
+}
+
+/// Translation of [`com.android.billingclient.api.SkuDetailsResponseListener`](https://developer.android.com/reference/com/android/billingclient/api/SkuDetailsResponseListener.html).
+///
+/// Returned as a datatype in a future instead of existing as a callback.
+class SkuDetailsResponseWrapper {
+  SkuDetailsResponseWrapper({@required this.responseCode, this.skuDetailsList});
+  static SkuDetailsResponseWrapper fromMap(Map<String, dynamic> map) {
+    final List<SkuDetailsWrapper> skuDetailsList = List.castFrom<dynamic,
```

Comment:
This PR has a lot of chains of `cast` calls like this to handle deserializing nested lists and maps from the channel. It looks pretty sad, I'd be happy for alternative suggestions. Without the numerous `cast`s this fails at runtime with type errors when trying to unbox a nested `List<>` or `Map<>`.

Reply:
Like we talked about offline, I decided to experiment with a code generator to take care of these boilerplate methods. Updated the PR description. It's called `json_serializable` and works around methods name `to/fromJson`, but works moving things to/from a `Map`. (Sorry for nitpicking about the terminology earlier @cyanglaz 😬)

Reply:
Never mind, split into a separate branch for a future PR like we talked about offline. For now this has been changed to take `Map<dynamic, dynamic>` to make casting a little simpler.